### PR TITLE
server: fix gh issue service error if there isnt repo field in issue

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Yet another regression found during local testing for #186. In some cases GitHub does not adding Repository field for issues in webhook requests. In those cases, we try to parse the `htmlURL` to get repo name and repo owner.

### CODE REVIEW

File: server/issue_handler_test.go [hunk 1]
```diff
@@ -96,6 +96,24 @@ func TestIssueEventHandler(t *testing.T) {
 		require.Equal(t, http.StatusBadRequest, resp.StatusCode)
 	})
 
+	t.Run("Should fail incorrect url", func(t *testing.T) {
```
Comment (agnivade) [inline server/issue_handler_test.go:99]: Can we also add a test where repo owner and name is nil but HTMLUrl is set, and verify that `ListLabelsByIssue` is being called with the correct values?

**Comment by agnivade** [comment]
Will do some testing with the other PRs.

**Comment by agnivade** [comment]
Tested and looks good!
